### PR TITLE
Drop redundant back/forward buttons

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
The path breadcrumb has back/forward buttons to its left which are essentially a redundant reimplementation of the browser back/forward buttons.  Let's drop that.

Dropping these will also clean up a not inconsiderable amount of the amount of state we're passing around between components.

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
Not having the buttons there really makes it clear how lonely the `/` item looks... weren't we doing to do something to help make that more clickable?

**Comment by jelly** [comment]
> Not having the buttons there really makes it clear how lonely the `/` item looks... weren't we doing to do something to help make that more clickable?

See this screenshot, it should be the name of your server.
image

**Comment by garrett** [comment]
I tried to check out this branch, but of course it won't work with fsinfo for me. I also couldn't get the VM working either. So I went back to the drawing board (literally) to doodle a bit and ended up with these revised mockups.

The icon probably needs to be a little larger (it's 24x16 here apparently).

Navigator, with inline breadcrumbs(1)

I also mocked up the topmost directory to show what it'd look like without the / in the pathbar. But there are weird side effects, especially with the / in the title. Should we say something else there? I tried a few things, but everything feels weird to me.

Navigator, with inline breadcrumbs(3)

Aside: I'm also wondering if we should recolor the link icons a little too. :thinking: 

If we copied the terminal, which looks like this:

image

...and used PF colors, it'd look like this:

Navigator, with inline breadcrumbs(5)

But people probably won't have that many symlinked folders and it's probably fine enough otherwise. :shrug: And this is too much like Christmas lights, and probably would look very odd with just 1 or 2 symlinks in a folder.

WDYT about all this? Especially the parts that are actually relevant to this PR. :sweat_smile:

**Comment by allisonkarlitskaya** [comment]
OK, so it's actually possible to delete ourselves.  That seems like a strange UI.  I think maybe we should drop it.

File: src/navigator-breadcrumbs.tsx [hunk 1]
```diff
@@ -18,17 +18,46 @@
...
+
     function navigate(n_parts: number) {
         cockpit.location.go("/", { path: encodeURIComponent(path.slice(0, n_parts).join("/")) });
     }
 
     return (
         <PageBreadcrumb stickyOnBreakpoint={{ default: "top" }}>
             <Flex spaceItems={{ default: "spaceItemsXs" }}>
+                <React.Fragment key="//hostname">
+                    <ServerIcon /> {hostname}
```
Comment (jelly) [inline src/navigator-breadcrumbs.tsx:59]: Confusingly you want to use `HddIcon`. Also this is not clickable while we want it to be. My patch for this. https://sprunge.us/odoJRK?javascript

**Comment by allisonkarlitskaya** [comment]
Rebased on main (after all the changes) and integrated your icon changes.

A couple of tests are failing, but should be easy enough to fix...

File: src/apis/spawnHelpers.jsx [hunk 2]
```diff
@@ -44,8 +44,6 @@ export const spawnDeleteItem = ({ path, selected, itemPath, Dialogs, setSelected
                     const newPath = path.slice(0, -1).join("/");
 
                     cockpit.location.go("/", { path: encodeURIComponent(newPath) });
```
[the remark this answers is not in the document]
Comment (jelly) [reply src/apis/spawnHelpers.jsx]: Sorry, I forgot to publish this comment while I was on the train. Locally I have created two test cases for this.

**Comment by jelly** [comment]
@allisonkarlitskaya pushed some (now) disabled tests for functionality which I think should work but we need still need to fix in the PR.

File: src/apis/spawnHelpers.jsx [hunk 3]
```diff
@@ -70,17 +63,15 @@ export const spawnForceDelete = ({ selected, path, Dialogs, setDeleteFailed, set
...
         ? path.slice(0, -1).join("/") + "/" + name
         : path.join("/") + "/" + name;
 
     cockpit.spawn(renameCommand({ selected, path, name }), options)
             .then(() => {
                 if (selected.items_cnt) {
-                    cockpit.location.go("/", { path: encodeURIComponent(newPath) });
-                    setHistory(h => h.slice(0, -1));
-                    setHistoryIndex(i => i - 1);
+                    cockpit.location.replace("/", { path: encodeURIComponent(newPath) });
```
Comment (jelly) [inline src/apis/spawnHelpers.jsx:74]: This is wrong for two reasons:

- should check for selected.dir as well, as that is set when you use the context menu
- somehow this changes the rename behavior to make a rename switch into the renamed dir

File: src/settings-dropdown.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+const _ = cockpit.gettext;
+
+export const SettingsDropdown = ({ showHidden, setShowHidden }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const onToggleClick = () => setIsOpen(!isOpen);
+    const onSelect = (_event, _itemId) => setIsOpen(false);
+
+    const onToggleHidden = () => {
+        setShowHidden(prevShowHidden => {
+            localStorage.setItem("cockpit-navigator.showHiddenFiles", !showHidden ? "true" : "false");
```
Comment (cockpituous) [inline src/settings-dropdown.jsx:42]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-234-20240212-174123-e2d87418-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]

**Pull request merged**